src/firefly/js/visualize/draw/ShapeToRegion.js

```diff
@@ -0,0 +1,663 @@
+    if (isEmpty(des)) return retList;
+
+    des += handleCommonRegionProps(color, lineWidth);
+
+    handleTextFromRegion(retList, des, drawObj, drawParams, RegionType.line, cc);
+
+    return retList;
+}
+
+/**
+ * ONLY VALID FOR CIRCLE!
+ * @param plot
+ * @param pts
+ * @return
+ */
+function getCircleCenter(plot,pts) {
+    const pt0= plot.getScreenCoords(pts[0]);
+    const pt1= plot.getScreenCoords(pts[1]);
+    if (!pt0 || !pt1) return null;
+    const x= Math.min(pt0.x,pt1.x) + Math.abs(pt0.x-pt1.x)/2;
+    const y= Math.min(pt0.y,pt1.y) + Math.abs(pt0.y-pt1.y)/2;
+    return plot.getWorldCoords(makeScreenPt(x,y));
+}
+
+/**
+ * calculate the radius on screen
+ * @param r
+ * @param plot
+ * @param unitType
+ * @returns {*}
+ */
+export function lengthToScreenPixel(r, plot, unitType) {
+    var screenRadius;
+
+    switch (unitType) {
+        case ShapeDataObj.UnitType.PIXEL:
+            screenRadius = r;
+            break;
+        case ShapeDataObj.UnitType.IMAGE_PIXEL:
+            screenRadius = (plot.zoomFactor * r);
+            break;
+        case ShapeDataObj.UnitType.ARCSEC:
+            screenRadius = getValueInScreenPixel(plot, r);
+            break;
+        default:
+            screenRadius = r;
+    }
+    return screenRadius;
+}
+
+/**
+ *
+ * @param plot
+ * @param pts
+ * @return {number}
+ */
+function findRadius(plot,pts) {
+    var retval= -1;
+    var pt0= plot.getScreenCoords(pts[0]);
+    var pt1= plot.getScreenCoords(pts[1]);
+    if (pt0 && pt1) {
+        var xDist= Math.abs(pt0.x-pt1.x)/2;
+        var yDist= Math.abs(pt0.y-pt1.y)/2;
+        retval= Math.min(xDist,yDist);
+    }
+    return retval;
+}
+
+/**
+ * create RegionValue (defined in Region.js) based on value and unit in terms of UnitType
+ * @param r
+ * @param unitType
+ */
+function makeRegionValue(r, unitType) {
+    var rV;
+
+    if (unitType === ShapeDataObj.UnitType.PIXEL) {
+        rV = RegionValue(r, RegionValueUnit.SCREEN_PIXEL);
+    } else if (unitType === ShapeDataObj.UnitType.IMAGE_PIXEL) {
+        rV = RegionValue(r, RegionValueUnit.IMAGE_PIXEL);
+    } else if (unitType === ShapeDataObj.UnitType.ARCSEC) {
+        rV = RegionValue(convertAngle('arcsec', 'deg', r), RegionValueUnit.DEGREE);
+    }
+
+    return rV;
+}
+
+/**
+ * create RegionDimension based on RegionValue in two dimensions in terms of UnitType
+ * @param w
+ * @param h
+ * @param unitType
+ * @returns {*}
+ */
+function makeRegionDim(w, h, unitType) {
+    return RegionDimension( makeRegionValue(w, unitType), makeRegionValue(h, unitType) );
+}
+
+/**
+ * create RegionValue based on angle and angleUnit, the return is in unit of degree
+ * @param cc
+ * @param angle
+ * @param angleUnit
+ * @returns {*}
+ */
+function makeRegionValueOnAngle(cc, angle, angleUnit) {
+    var rAngle = 0.0;
+
+    if (!isNil(angle) && angleUnit) {
+        if (angleUnit === ShapeDataObj.UnitType.ARCSEC) {
+            rAngle = convertAngle('arcsec', 'deg', angle);
```

Comment:
is convertAngle new it VisUtil?  I don't think I have seen it before. Would you mind putting some JSDocs on it?